### DISCUSSION TITLE
[Idea] ModuleFrontController accessible even if the module is disabled

### DISCUSSION THREAD

**Discussion by kpodemski** [PrestaShop/PrestaShop#35387] [Ideas]
### The problem

Following this PR https://github.com/PrestaShop/PrestaShop/pull/34184 we've had to make some adjustments to some of the native modules, for instance: https://github.com/PrestaShop/ps_facetedsearch/pull/983, even though it solves the original problem, it creates another one. The ModuleFrontController is not accessible if the module is disabled, the old files were accessible, so you could index the products before enabling the module.

This might be a manageable problem for some of the modules, but it could be for some.

The check is here:
https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Dispatcher.php#L386
and here:
https://github.com/PrestaShop/PrestaShop/blob/develop/classes/controller/ModuleFrontController.php#L38,

### The solution

The solution would be to create a new property or method allowing the developer to prepare a list of controllers that should be accessible even if the module is disabled.

For instance:

```
class MyModule extends Module
{
    ...

    public function getUnrestrictedControllers(): array
    {
        return [
            'cron',
        ];
    }
}
```

We could later respect these controllers not to restrict them if module is disabled.

### Alternatives

@Hlavtox has some alternative proposals

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by Hlavtox** [comment]
I would go for a `$allowWhenDisabled = false` property to a `Controller` class.

Core would check it when routing and instantiating this controller. If true, it would run the controller even if `$module->active != 1`.

**Reply by kpodemski** [reply]
yes, that could also work but it would complicate the check inside the Dispatcher, you would have to use Reflection to get this property value

**Reply by kpodemski** [reply]
well, we already do that for other properties, so maybe it does make sense

https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Dispatcher.php#L1245

**Reply by Hlavtox** [reply]
Or a static property?
